src/main.py

```diff
@@ -109,20 +115,68 @@ def display_channel_to_article_type(display_channel_list):
     "http://creativecommons.org/publicdomain/zero/1.0/": "CC0-1.0"
 }
 
+def identity(x):
+    return x
+
+def related_article_to_reviewed_preprint(soup):
+    """returns a list of /reviewed-preprint snippets for any related article detected as using one.
+    detection involves inspecting article references for 'RP' prefixed manuscript ids.
+    """
+    pub_date = parseJATS.pub_date(soup)
+    pub_date = to_datetime(pub_date)
+    if not pub_date:
+        return []
+
+    if rpp.before_inception(pub_date):
+        return []
+
+    def fetch(msid):
+        return rpp.snippet(msid)
+
+    def msid_from_relation(struct):
+        return utils.msid_from_elife_doi(struct.get('xlink_href'))
+
+    related_article_list = parseJATS.related_article(soup)
+    msid_list = list(map(msid_from_relation, related_article_list))
+
+    # brute force approach. check API for every related MSID.
+    # return list(filter(None, map(fetch, reference_msid_list)))
+
+    if not msid_list:
+        return []
+
+    # `elifetools` is modifying the soup as we access it.
+    # multiple accesses to `references_json` results in strange behaviour,
+    # it's also hugely *slow*.
+    # * 24271 v1, ref 'bib38': the 'date' disappears.
+    # copy the soup before accessing `references_json`
+    references = parseJATS.references_json(copy.copy(soup))
+
+    # check each reference for any relation's msid prefixed with an 'RP'
+    reference_msid_list = []
+    for ref in references:
+        val = ref.get('pages')
+        # `val` may be a string or a dict. just check strings for now.
+        if val and isinstance(val, str) and val[:2] == "RP":
```

Comment:
Probably a rare situation, a non-eLife article may have an elocation-id starting with `RP`

Reply:
Although if is also must have the right `msid` it'll probably be fine